src/huggingface_hub/file_download.py

```diff
@@ -142,6 +147,9 @@ def get_fastcore_version():
     return _fastcore_version
 
 
+REGEX_COMMIT_HASH = re.compile(r"^[0-9a-f]{5,40}$")
```

Comment:
Theoretically it could be 4 instead of 5 here, as hf.co resolves correctly down to as low as 4 characters
```suggestion
REGEX_COMMIT_HASH = re.compile(r"^[0-9a-f]{4,40}$")
```

Reply:
(right now putting 4 characters means that we're skipping the cache check and going straight for `hf_hub_url`)

Reply:
oh i actually meant to only allow passing a full commit (i.e. just `{40}`), to make sure no false positive are detected (like naming a branch `deadbeef` etc)

Do you see an issue in accepting only full commit hashes?